spec/lib/payment_calculator/ecf/payment_calculation_spec.rb

```diff
@@ -26,25 +26,32 @@
 
   it "returns the expected types for all outputs" do
     @combined_results = nil
-    retained_event_aggregations.each do |key, value|
+    retained_event_aggregations.each_with_index do |(key, value), index|
       result = described_class.call(lead_provider: contract.lead_provider, event_type: key, total_participants: value)
 
       if @combined_results.nil?
         expect(result.dig(:service_fees, :service_fee_per_participant)).to be_a(BigDecimal)
         expect(result.dig(:service_fees, :service_fee_total)).to be_a(BigDecimal)
         expect(result.dig(:service_fees, :service_fee_monthly)).to be_a(BigDecimal)
-        expect(result.dig(:output_payment, :per_participant)).to be_a(BigDecimal)
+        expect(result.dig(:output_payments, 0, :per_participant)).to be_a(BigDecimal)
+        expect(result.dig(:output_payments, 1, :per_participant)).to be_a(BigDecimal)
+        expect(result.dig(:output_payments, 2, :per_participant)).to be_a(BigDecimal)
       end
 
-      @combined_results ||= { service_fees: result[:service_fees], output_payment: {} }
-      @combined_results[:output_payment][key] = result.dig(:output_payment, key)
-    end
+      @combined_results ||= { service_fees: result[:service_fees], output_payments: result[:output_payments] }
+
+      unless @combined_results[:output_payments][index].nil?
+        @combined_results[:output_payments][index][key] = result.dig(:output_payments, index, key)
+      end
 
-    @combined_results[:output_payment].each do |key, value|
       expect(key).to be_a(String)
-      expect(value[:retained_participants]).to be_an(Integer)
-      expect(value[:per_participant]).to be_a(BigDecimal)
-      expect(value[:subtotal]).to be_a(BigDecimal)
+      @combined_results[:output_payments].each_with_index do |data, _index|
+        next if data[key].nil?
+
+        expect(data[key][:retained_participants]).to be_an(Integer)
+        expect(data[key][:per_participant]).to be_a(BigDecimal)
+        expect(data[key][:subtotal]).to be_a(BigDecimal)
+      end
```

Comment:
If we are not using the `index`, let's use `each` rather than `each_with_index`.